src/coreclr/jit/valuenum.cpp

```diff
@@ -3012,7 +3012,7 @@ ValueNum ValueNumStore::EvalCastForConstantArgs(var_types typ, VNFunc func, Valu
         case TYP_FLOAT:
         {
             float arg0Val = GetConstantSingle(arg0VN);
-            assert(!checkedCast || !CheckedOps::CastFromFloatOverflows(arg0Val, castToType));
+            assert(!CheckedOps::CastFromFloatOverflows(arg0Val, castToType));
```

Comment:
Could you confirm why was `checkedCast` removed from the assertion?

Reply:
The assertion used to test: `if (checkedCast) { assert(noOverflow) }`, now it tests that there can be no overflow, even for regular casts. This is double checking that `VNEvalShouldFold` is not letting overflowing casts though.